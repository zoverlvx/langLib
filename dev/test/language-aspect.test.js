import React from "react";
import LanguageAspect from "../js/components/language-aspect";
import toJson from "enzyme-to-json";
import {shallow} from "enzyme";

const component = shallow(<LanguageAspect />);
const tree = toJson(component);

describe("<LanguageAspect />", () => {
    const minProps = {
    	className: "language_aspect",
    	key: {},
    	onClick: () => {}
     	}

    it("renders 1 <LanguageAspect /> component", () => {
		console.log(tree); 
        expect(component).toHaveLength(1);
    });

    it("provides a snapshot of LanguageAspect's tree", () => { 
	    expect(tree).toMatchSnapshot();
	});

	it("renders props correctly", () => {
        expect(
         	shallow(
         		<LanguageAspect {...minProps} />
         	).length 
         ).toEqual(1); 
     });
});

//What ought to be tested





