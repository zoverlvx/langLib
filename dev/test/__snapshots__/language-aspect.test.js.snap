// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageAspect /> provides a snapshot of LanguageAspect's tree 1`] = `
<Route
  render={[Function]}
/>
`;
